src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx

```diff
@@ -426,19 +439,33 @@ const TableBodyRow = ({
             align={align}
             className={`data-table-${align}`}
           >
-            <StyledNestedSpan nestingOffset={offset}>
-              {col.renderDataFunction ? (
-                col.renderDataFunction({
-                  // Call the column renderer if it's provided
-                  value: values[col.columnId],
-                  columnId: col.columnId,
-                  rowId: id,
-                  row: values,
-                })
-              ) : (
-                <TableCellRenderer>{values[col.columnId]}</TableCellRenderer>
-              )}
-            </StyledNestedSpan>
+            {editBar !== 'buttons' ? (
+              <StyledNestedSpan nestingOffset={offset}>
+                {col.renderDataFunction ? (
+                  col.renderDataFunction({
+                    // Call the column renderer if it's provided
+                    value: values[col.columnId],
+                    columnId: col.columnId,
+                    rowId: id,
+                    row: values,
+                  })
+                ) : (
+                  <TableCellRenderer>{values[col.columnId]}</TableCellRenderer>
+                )}
+              </StyledNestedSpan>
+            ) : (
+              <StyledNestedSpan nestingOffset={offset}>
+                {col.editDataFunction ? (
+                  col.editDataFunction({
+                    value: values[col.columnId],
+                    columnId: col.columnId,
+                    rowId: id,
```

Comment:
Give the editDataFunction the same signature of the renderDataFunction (ie, same 4 paramters).  Otherwise I'll be having to read the code to figure out why, in 3 months, the editDataFunction is not accepting a row.

Reply:
In my opinion, the default value of `editBar` is `null`, and when clicked edit icon in toolbar, set `editBar` to `buttons` (i.e. save and cancel buttons showed up). When clicked save button, set `editBar ` from `buttons ` to `toast` (i.e. the toast showed up). Or when clicked cancel button, set `editBar ` from `buttons ` to `null` (i.e. toolbar showed up with search, filter, edit icons, etc.). Besides, when clicked the toast, set `editBar ` from `toast ` to `null` (i.e. toolbar showed up with search, filter, edit icons, etc.).

Reply:
got it, I'll give the editDataFunction the same signature of the renderDataFunction.